src/test/java/teammates/test/cases/logic/CommentsLogicTest.java

```diff
@@ -370,6 +370,28 @@ public void testUpdateComment() throws Exception {
         assertEquals(1, actual.size());
         assertEquals(c.commentText, actual.get(0).commentText);
     }
+    
+    @Test
+    public void testUpdateCommentEmail() throws Exception {
+        CommentAttributes existingComment1 = dataBundle.comments.get("comment1FromI1C1toS1C1");
+        
+        ______TS("typical success case: email changed");
+        String courseId = existingComment1.courseId;
+        String oldInstrEmail = existingComment1.lastEditorEmail;
+        String updatedInstrEmail = "otheremail@gmail.tmt";
+        commentsLogic.updateInstructorEmail(courseId, oldInstrEmail, updatedInstrEmail);
+        
+        List<CommentAttributes> updatedComments = commentsLogic.getCommentsForGiver(courseId, updatedInstrEmail);
+        for (CommentAttributes updatedComment : updatedComments) {
+            assertEquals(updatedInstrEmail, updatedComment.giverEmail);
+            if (updatedComment.lastEditorEmail != null) {
+                assertEquals(updatedInstrEmail, updatedComment.lastEditorEmail);
+            }
+        }
+        
+        // edit back
+        commentsLogic.updateInstructorEmail(courseId, updatedInstrEmail, oldInstrEmail);
```

Comment:
It's persisted at the start of class, but we could easily change it to be the start of every test. We have many violations of this, but I think we should try to avoid having code to tear down or setup in a method annotated with `@Test`, ideally tests shouldn't need depend on each other to have the right state.